central/complianceoperator/v2/report/manager/complianceReportgenerator/report_gen_impl.go

```diff
@@ -84,88 +92,151 @@ type ResultEmail struct {
+
 	formatEmailBody := &formatBody{
 		BrandedPrefix: branding.GetCombinedProductAndShortName(),
-		Profile:       profiles,
+		Profile:       strings.Join(req.Profiles, ","),
 		Pass:          data.TotalPass,
 		Fail:          data.TotalFail,
 		Mixed:         data.TotalMixed,
 		Cluster:       len(req.ClusterIDs),
 	}
 
-	if len(profiles) > maxNumberProfilesinSubject {
-		profiles = strings.Join(req.Profiles[0:maxNumberProfilesinSubject], ", ")
-		profiles += "..."
-	}
-
 	formatEmailSub := &formatSubject{
 		BrandedPrefix: branding.GetCombinedProductAndShortName(),
 		ScanConfig:    req.ScanConfigName,
-		Profiles:      len(profiles),
+		Profiles:      len(req.Profiles),
 	}
 
 	log.Infof("Sending email for scan config %s", req.ScanConfigName)
 	go rg.sendEmail(req.Ctx, zipData, formatEmailBody, formatEmailSub, req.Notifiers)
 }
 
-// getDataForReport returns map of cluster id and
-func (rg *complianceReportGeneratorImpl) getDataForReport(req *ComplianceReportRequest) (*ResultEmail, error) {
-	// TODO ROX-24356: Implement query to get checkresults data to generate cvs for compliance reporting
+// getDataforReport returns map of cluster id and slice of ResultRow
+func (rg *complianceReportGeneratorImpl) getDataforReport(req *ComplianceReportRequest) (*ResultEmail, error) {
+	clusters := req.ClusterIDs
+	resultsCSV := make(map[string][]*ResultRow)
+	resultEmailComplianceReport := &ResultEmail{
+		TotalPass:  0,
+		TotalMixed: 0,
+		TotalFail:  0,
+		Clusters:   0,
+	}
 
-	return nil, nil
+	for _, clusterID := range clusters {
+		parsedQuery := search.NewQueryBuilder().AddExactMatches(search.ComplianceOperatorScanConfig, req.ScanConfigID).
+			AddExactMatches(search.ClusterID, clusterID).
+			ProtoQuery()
+		resultCluster := []*ResultRow{}
+
+		err := rg.checkResultsDS.WalkByQuery(req.Ctx, parsedQuery, func(c *storage.ComplianceOperatorCheckResultV2) error {
+			row := &ResultRow{
+				ClusterName: c.GetClusterName(),
+				CheckName:   c.GetCheckName(),
+				Description: c.GetDescription(),
+				Status:      c.GetStatus().String(),
+				Remediation: c.GetInstructions(),
+			}
+			// get profile name for the check result
+			q := search.NewQueryBuilder().AddExactMatches(search.ComplianceOperatorScanRef, c.GetScanRefId()).ProtoQuery()
+			log.Info("query to get profiles is %s", q)
+			profiles, err := rg.profileDS.SearchProfiles(req.Ctx, q)
+			if err != nil {
+				return err
+			}
+			if len(profiles) < 1 {
+				return errors.New("Profile not found")
+			}
+
+			row.Profile = profiles[0].GetName()
+
+			// get remediation for the check result
+			q = search.NewQueryBuilder().AddExactMatches(search.ComplianceOperatorCheckName, c.GetCheckName()).AddExactMatches(search.ClusterID, c.GetClusterId()).ProtoQuery()
+			remediations, err := rg.remediationDS.SearchRemediations(req.Ctx, q)
+			if err != nil {
+				return err
+			}
+			if len(remediations) > 0 {
+				row.Remediation = remediations[0].GetName()
+			}
```

Comment:
@SimonBaeumer is it possible to have more than one remediation for a given check result on a given cluster?

Reply:
Yes, it is. For example it looks like this:
```
$ oc get rems  | grep network-policy
ocp4-stig-project-config-and-template-network-policy                                            NotApplied
ocp4-stig-project-config-and-template-network-policy-1                                          NotApplied
```